### PR TITLE
Update Package: py-pre-commit

### PR DESCRIPTION
**Pull request by jcpunk**
This adds the latest version of pre-commit to the package list.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class PyPreCommit(PythonPackage):
     homepage = "https://github.com/pre-commit/pre-commit"
     pypi = "pre_commit/pre_commit-1.20.0.tar.gz"
 
+    version('2.17.0', sha256='c1a8040ff15ad3d648c70cc3e55b93e4d2d5b687320955505587fd79bbaed06a')
     version('1.20.0', sha256='9f152687127ec90642a2cc3e4d9e1e6240c4eb153615cb02aa1ad41d331cbb6e')
 
     depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py:19]: Latest version requires Python 3.6.1+

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 2]
```diff
@@ -22,9 +23,11 @@ class PyPreCommit(PythonPackage):
     depends_on('py-identify@1.0.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py:24]: py-aspy-yaml is no longer required

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 3]
```diff
@@ -22,9 +23,11 @@ class PyPreCommit(PythonPackage):
...
     depends_on('py-nodeenv@0.11.1:', type=('build', 'run'))
     depends_on('py-pyyaml', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1:', type=('build', 'run'), when="@2.1.0:")
     depends_on('py-six', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py]: py-six is no longer needed

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 4]
```diff
@@ -22,9 +23,11 @@ class PyPreCommit(PythonPackage):
...
     depends_on('py-toml', type=('build', 'run'))
     depends_on('py-virtualenv@15.2:', type=('build', 'run'))
+    depends_on('py-virtualenv@20.0.8:', type=('build', 'run'), when="@2.4.0:")
     depends_on('py-futures', type=('build', 'run'), when='^python@:3.1')
     depends_on('py-importlib-metadata', type=('build', 'run'), when='^python@:3.7')
     depends_on('py-importlib-resources', type=('build', 'run'), when='^python@:3.6')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py]:
```suggestion
    depends_on('py-importlib-resources@:5.2', type=('build', 'run'), when='^python@:3.6')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by jcpunk** [comment]
I believe I've resolved the requested bits

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 5]
```diff
@@ -13,18 +13,22 @@ class PyPreCommit(PythonPackage):
...
     depends_on('py-setuptools', type='build')
-    depends_on('py-aspy-yaml', type=('build', 'run'))
+    depends_on('py-aspy-yaml', type=('build', 'run'), when="@:2")
     depends_on('py-cfgv@2.0.0:', type=('build', 'run'))
     depends_on('py-identify@1.0.0:', type=('build', 'run'))
     depends_on('py-nodeenv@0.11.1:', type=('build', 'run'))
     depends_on('py-pyyaml', type=('build', 'run'))
-    depends_on('py-six', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1:', type=('build', 'run'), when="@2.1.0:")
+    depends_on('py-six', type=('build', 'run'), when="@:2")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py]: `@:2` means all versions up to and including 2.X.Y. Maybe you want `@1` (all 1.X.Y versions) or `@:2.1` (all versions up to and including 2.1.Y?

File: var/spack/repos/builtin/packages/py-pre-commit/package.py [hunk 6]
```diff
@@ -13,18 +13,22 @@ class PyPreCommit(PythonPackage):
...
     pypi = "pre_commit/pre_commit-1.20.0.tar.gz"
 
+    version('2.17.0', sha256='c1a8040ff15ad3d648c70cc3e55b93e4d2d5b687320955505587fd79bbaed06a')
     version('1.20.0', sha256='9f152687127ec90642a2cc3e4d9e1e6240c4eb153615cb02aa1ad41d331cbb6e')
 
     depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
+    depends_on('python@3.6.1:', type=('build', 'run'), when="@2.1.0:")
     depends_on('py-setuptools', type='build')
-    depends_on('py-aspy-yaml', type=('build', 'run'))
+    depends_on('py-aspy-yaml', type=('build', 'run'), when="@:2")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pre-commit/package.py]: `@:2` means all versions up to and including 2.X.Y. Maybe you want `@1` (all 1.X.Y versions) or `@:2.1` (all versions up to and including 2.1.Y?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pre-commit/package.py (hunk 5 above)
Comment (jcpunk) [reply var/spack/repos/builtin/packages/py-pre-commit/package.py]: I'll fix this shortly

File: var/spack/repos/builtin/packages/py-pre-commit/package.py (hunk 6 above)
Comment (jcpunk) [reply var/spack/repos/builtin/packages/py-pre-commit/package.py]: I'll fix this shortly

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**